Frontend/android-native/mobility-app/src/main/java/in/juspay/mobility/app/OverlaySheetService.java

```diff
@@ -119,14 +119,36 @@ public static void deRegisterCallback(CallBack notificationCallback) {
     @SuppressLint("SetTextI18n")
     private void updateTipView (SheetAdapter.SheetViewHolder holder, SheetModel model) {
         mainLooper.post(() -> {
-            if (model.getCustomerTip() > 0 || model.getDisabilityTag()) {
+            String variant = model.getRequestedVehicleVariant();
+            if (model.getCustomerTip() > 0 || model.getDisabilityTag() || (!variant.equals("") && key != null && key.equals("yatrisathiprovider"))) {
                 holder.customerTipBlock.setVisibility(View.VISIBLE);
                 if (model.getDisabilityTag()) {
                     holder.accessibilityTag.setVisibility(View.VISIBLE);
                     holder.accessibilityTagText.setText(getString(R.string.assistance_required));
                 }  else{
                     holder.accessibilityTag.setVisibility(View.GONE);
                 }
+
+                if (variant.equals("") && key != null && key.equals("yatrisathiprovider")) {
```

Comment:
Why are we checking `variant.equals("")` ?

Reply:
refactored function